lib/buffer.js

```diff
@@ -66,10 +72,67 @@ function Buffer(arg, encoding) {
   return fromObject(arg);
 }
 
+/**
+ * Functionally equivalent to Buffer(arg, encoding) but throws a TypeError
+ * if value is a number.
+ **/
+Buffer.from = function(value, encoding) {
+  if (typeof value === 'number')
+    throw new TypeError(`'value' argument must not be a number`);
+  return typeof value === 'string' ?
+    fromString(value, encoding) :
+    fromObject(value);
```

Comment:
style nit: statement continuations indented twice. (going to check if we can add that to the linter)